projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/config.hpp

```diff
@@ -131,6 +131,10 @@ struct config : output_config
     bool   list_metrics                = get_env("ROCPROF_LIST_METRICS", false);
     bool   list_metrics_output_file    = get_env("ROCPROF_OUTPUT_LIST_METRICS_FILE", false);
     bool   advanced_thread_trace       = get_env("ROCPROF_ADVANCED_THREAD_TRACE", false);
+    bool   spm_counter_collection      = get_env("ROCPROF_SPM_COUNTER_COLLECTION", false);
+    size_t spm_buffer_size             = get_env("ROCPROF_SPM_BUFFER_SIZE", 32768);
+    size_t spm_timeout_ms              = get_env("ROCPROF_SPM_TIMEOUT_MS", 50);
+    size_t spm_frequency_sclk          = get_env("ROCPROF_SPM_FREQUENCY", 500000);
```

Comment:
You really should define these variable with the units, e.g. `spm_frequency_sclk_mhz`